_docker/drupal/drupal-filesystem/web/modules/custom/rhd_common/rhd_common.sitemap.inc

```diff
@@ -39,9 +39,58 @@ function product_node_sitemap_entries(&$links) {
   // this add all the sub pages to the list of links and unsets the one with the node entry.
   $callback = function ($product_node_link, $product_node_link_index) use (&$links) {
     $product = Node::load($links[$product_node_link_index]['meta']['entity_info']['id']);
-    $sub_pages = $product->field_product_pages->referencedEntities();
-    $entries = array_map('build_subpage_link_entry', $sub_pages,
-      array_fill(0, count($sub_pages), $links[$product_node_link_index]));
+
+    // The product page is using the new, assembly-based approach.
+    if ($product->hasField('field_use_new_product_page')
+      && !empty($product->field_use_new_product_page->value)
+      && $product->field_use_new_product_page->value == '1') {
+      /** @var \Drupal\Core\Config\ImmutableConfig $rhd_config */
+      $rhd_drupal_config = \Drupal::config('redhat_developers')->get('drupal');
```

Comment:
@jordanpagewhite I'm not totally up on variable scoping in PHP lambdas, but it seems to me we only need to generate the base URL once per cron run?